src/main/cc/proxy_client/proxy_client.cc

```diff
@@ -157,6 +162,36 @@ int GetInputsFromIncludeProcessor(const string& cmd_id, int argc, char** argv, c
   return 0;
 }
 
+
+string readFile(const std::string& fileName) {
+  ifstream fileStream(fileName.c_str());
+  return string((istreambuf_iterator<char>(fileStream)),
+            istreambuf_iterator<char>());
+}
+
+bool isFileArg(const std::string& str) {
+  if (str.substr(0, 5) == FILE_ARG_PREFIX) {
+    return true;
+  }
+  return false;
+}
+
+void expandFileArguments(set<string>* args) {
+  set<string> argsFromFileContents;
+
+  for (const auto& arg : *args) {
+    if (isFileArg(arg)) {
+      const std::string& fileName = arg.substr(5);
+      const std::string& fileContents = readFile(fileName);
+      for (const auto& argFromFile : absl::StrSplit(fileContents, ',', absl::SkipEmpty())) {
+        argsFromFileContents.insert(string(argFromFile));
```

Comment:
Is this cast necessary? It seems like StrSplit returns a vector of strings.

Reply:
StrSplit returns a string_view which does not implicitly cast to string (I got a compile error). Hence the cast here